### PR TITLE
Create Posts with a specified language, or fallback

### LINKED ISSUE #781: 🚀 [Feature] Language of Contribution
> [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-06-07T11:06:10Z" title="Friday, June 7th 2019, 1:06:10 pm +02:00">Jun 7, 2019</time>_
_Closed <time datetime="2019-06-14T16:53:23Z" title="Friday, June 14th 2019, 6:53:23 pm +02:00">Jun 14, 2019</time>_
---

## :rocket: Feature

The User should be able to select the Language of the Contribution.

### Design & Layout

Alpha - Create Post - Language - Bildschirmfoto 2019-06-06 um 09 13 14

### PR DESCRIPTION
**Pull request by Tirokk**
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-06-10T20:13:17Z" title="Monday, June 10th 2019, 10:13:17 pm +02:00">Jun 10, 2019</time>_
_Merged <time datetime="2019-06-14T16:53:22Z" title="Friday, June 14th 2019, 6:53:22 pm +02:00">Jun 14, 2019</time>_
---

### Issues

- fixes #781 

### Todo

- [ ] Write component tests for languages

### CODE REVIEW

**Comment by Tirokk** [comment]
> [codecov[bot]](https://github.com/apps/codecov) **Authored by [codecov[bot]](https://github.com/apps/codecov)**
_<time datetime="2019-06-11T18:47:57Z" title="Tuesday, June 11th 2019, 8:47:57 pm +02:00">Jun 11, 2019</time>_

---


# [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/795?src=pr&el=h1) Report
> :exclamation: No coverage uploaded for pull request base (`master@14041ff`). [Click here to learn what that means](https://docs.codecov.io/docs/error-reference#section-missing-base-commit).
> The diff coverage is `n/a`.

**Comment by Tirokk** [comment]
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-06-12T22:46:12Z" title="Thursday, June 13th 2019, 12:46:12 am +02:00">Jun 13, 2019</time>_

---


Screenshot at 2019-06-12 19:40:21
Screenshot at 2019-06-12 19:40:21
Screenshot at 2019-06-12 19:42:19

**Comment by Tirokk** [comment]
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-06-13T23:49:04Z" title="Friday, June 14th 2019, 1:49:04 am +02:00">Jun 14, 2019</time>_

---


Screenshot at 2019-06-13 17:27:40

this is what you would expect @Tirokk??

File: webapp/components/ContributionForm/index.vue [hunk 1]
```diff
@@ -6,8 +6,27 @@
...
+          <ds-flex-item :width="{ base: '80%', sm: '30%', md: '30%', lg: '20%' }">
+            <ds-space margin-bottom="small" />
+            <ds-select
+              model="language"
+              :options="form.languageOptions"
+              icon="globe"
+              :placeholder="form.placeholder"
+              :label="$t('contribution.languageSelectLabel')"
+            />
+          </ds-flex-item>
```
Comment (Tirokk) [inline webapp/components/ContributionForm/index.vue:21]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-06-14T07:21:17Z" title="Friday, June 14th 2019, 9:21:17 am +02:00">Jun 14, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -137,7 +138,10 @@ export default {
       this.$refs.contributionForm.update('content', value)
     },
     returnLocaleName(locale) {
-      if (this.$i18n.locale() === locale.code) {
+      if (
+        (this.contribution && this.contribution.language === locale.code) ||
+        (!this.contribution && this.$i18n.locale() === locale.code)
+      ) {
```


What happens in the else case?

Comment (Tirokk) [reply webapp/components/ContributionForm/index.vue:21]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-06-14T10:52:34Z" title="Friday, June 14th 2019, 12:52:34 pm +02:00">Jun 14, 2019</time>_

---


good question @Tirokk, I didn't think there was a case where `this.$i8n.locale()` is undefined :thinking: 
what would you expect to happen if there is no locale?

Comment (Tirokk) [reply webapp/components/ContributionForm/index.vue:21]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-06-14T11:23:04Z" title="Friday, June 14th 2019, 1:23:04 pm +02:00">Jun 14, 2019</time>_

---


Just use english as the world language @mattwr18 , I think. Or?
If the else case is not possible, we don’t need an ìf` here?!

Comment (Tirokk) [reply webapp/components/ContributionForm/index.vue:21]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-06-14T16:47:24Z" title="Friday, June 14th 2019, 6:47:24 pm +02:00">Jun 14, 2019</time>_

---


Addressed in our latest pairing session @Tirokk... unfortunately, I forgot to add you as a co-author, but I think you know I appreciate you!!

**Review by Mogge — APPROVED** [review]

**Review by Mogge — CHANGES_REQUESTED** [review]
There are some function problems:
- If I am in german as a user and I write a post in english, save and edit again, the language is shown as german.
- If I edit the Post again, change the language and save again, the language seems to be still my own and not the language I set.
- Discussable: The language is not shown in the Post, if I don’t edit.

**Pull request closed without merging**